src/http-client.spec.ts

```diff
@@ -0,0 +1,84 @@
+import FetchHttpClient, { HttpRequestError, ISdkParams } from './http-client';
+
+describe('FetchHttpClient', () => {
+  const baseUrl = 'http://api.example.com';
+  const sdkParams: ISdkParams = {
+    apiKey: '12345',
+    sdkVersion: '1.0',
+    sdkName: 'ExampleSDK',
+  };
+  const timeout = 5000; // 5 seconds
+
+  let httpClient: FetchHttpClient;
+
+  beforeEach(() => {
+    httpClient = new FetchHttpClient(baseUrl, sdkParams, timeout);
+    global.fetch = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('should return data successfully when HTTP status is 200', async () => {
+    const mockJsonPromise = Promise.resolve({ data: 'test' });
+    const mockFetchPromise = Promise.resolve({
+      ok: true,
+      json: () => mockJsonPromise,
+      status: 200,
+    });
+    (global.fetch as jest.Mock).mockImplementation(() => mockFetchPromise);
+
+    const resource = '/data';
+    const result = await httpClient.get(resource);
+
+    expect(global.fetch).toHaveBeenCalledTimes(1);
+    expect(global.fetch).toHaveBeenCalledWith(
+      `${baseUrl}${resource}?apiKey=12345&sdkVersion=1.0&sdkName=ExampleSDK`,
+      { signal: expect.any(AbortSignal) },
+    );
+    expect(result).toEqual({ data: 'test' });
+  });
+
+  it('should throw HttpRequestError when HTTP status is not 200', async () => {
+    const mockFetchPromise = Promise.resolve({
+      ok: false,
+      status: 404,
+      statusText: 'Not Found',
+    });
+    (global.fetch as jest.Mock).mockImplementation(() => mockFetchPromise);
+
+    const resource = '/data';
+    await expect(httpClient.get(resource)).rejects.toThrow(HttpRequestError);
+    await expect(httpClient.get(resource)).rejects.toEqual(
+      new HttpRequestError('Failed to fetch data', 404),
+    );
+  });
+
+  it('should throw HttpRequestError on timeout', async () => {
+    jest.useFakeTimers();
```

Comment:
PTSD triggered 😅 

https://docs.google.com/presentation/d/1Wi5q3iAQHY60xepgLuv0PYjYnPojCrIvO0bOIyStuKM/edit#slide=id.g2b3575c090a_0_0

Reply:
You made my work easy!